From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Subject: tools/power/x86/intel-speed-select: Fix TRL restore after SST-TF disable
References: PED-14190
Patch-Mainline: v6.14-rc1
Git-commit: 991c8aacfb6e3088027b8c776ad31d2c093905b8


When SST-TF is disabled, the TRL (Turbo Ratio Limit) of config level 0
is getting restored. But the TRL of current level should be restored
which may not be config level 0.

This is caused by a bug in treating config level as TRL level. So
arguments needs to be swapped.

Signed-off-by: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>


Signed-off-by:  <trenn@suse.de>
---
 tools/power/x86/intel-speed-select/isst-core-tpmi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tools/power/x86/intel-speed-select/isst-core-tpmi.c
+++ b/tools/power/x86/intel-speed-select/isst-core-tpmi.c
@@ -329,7 +329,7 @@
 	return 0;
 }
 
-static int tpmi_get_get_trl(struct isst_id *id, int level, int config_index,
+static int tpmi_get_get_trl(struct isst_id *id, int config_index, int level,
 			    int *trl)
 {
 	struct isst_pkg_ctdp_level_info ctdp_level;
